src/backend/distributed/utils/citus_outfuncs.c

```diff
@@ -479,9 +485,7 @@ OutTask(OUTFUNC_ARGS)
 	WRITE_ENUM_FIELD(taskType, TaskType);
 	WRITE_UINT64_FIELD(jobId);
 	WRITE_UINT_FIELD(taskId);
-	WRITE_NODE_FIELD(queryForLocalExecution);
-	WRITE_STRING_FIELD(queryStringLazy);
-	WRITE_NODE_FIELD(queryStringList);
+	WRITE_NODE_FIELD(taskQuery);
```

Comment:
does this actually work in all cases? (e.g. even when it's a string?)

Reply:
I think this will call `OutTaskQuery`. However we should output the union in `OutTaskQuery` with a switch case I believe. Something like this: https://github.com/citusdata/citus/pull/3659/files/0163c66241c9b7a56db82ed0272f2d9305c8f295..fff4f81fed5058527c982e6b2b08a588c17af137#diff-1ea4be14ccb6a4b95fab5db3e9c18e90R259